### PR TITLE
8338409: Use record to simplify code

### PR DESCRIPTION
**Pull request by wenshao**
j.u.Formatter$FixedString can be refactored to simplify the code using Record

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8338409](https://bugs.openjdk.org/browse/JDK-8338409): Use record to simplify code (**Enhancement** - P4)

### Reviewers
 * [Claes Redestad](https://openjdk.org/census#redestad) (@cl4es - **Reviewer**)
 * [Chen Liang](https://openjdk.org/census#liach) (@liach - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20443/head:pull/20443` \
`$ git checkout pull/20443`

Update a local copy of the PR: \
`$ git checkout pull/20443` \
`$ git pull https://git.openjdk.org/jdk.git pull/20443/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20443`

View PR using the GUI difftool: \
`$ git pr show -t 20443`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20443.diff">https://git.openjdk.org/jdk/pull/20443.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20443#issuecomment-2290032823)

### CODE REVIEW

File: src/java.base/share/classes/java/util/Formatter.java [hunk 1]
```diff
@@ -3021,15 +3021,7 @@ interface FormatString {
...
-    private static class FixedString implements FormatString {
-        private final String s;
-        private final int start;
-        private final int end;
-        FixedString(String s, int start, int end) {
-            this.s = s;
-            this.start = start;
-            this.end = end;
-        }
+    private static record FixedString(String s, int start, int end) implements FormatString {
```
Comment (liach) [inline src/java.base/share/classes/java/util/Formatter.java:3024]:
```suggestion
    private record FixedString(String s, int start, int end) implements FormatString {
```
Records are always implicitly static.

**Review by cl4es — APPROVED** [review]

**Review by liach — APPROVED** [review]
A simple cleanup.

**Pull request closed without merging**